### PR TITLE
Adding a CLI, and batch processing.

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

**Comment by vankasteelj** [comment]
I cannot merge something with so many undocumented changes and a name change as well

**Comment by dessalines** [comment]
Gotcha, sry lemme fix that. I'll make some comments too on the changes, and make sure you agree with em.

File: README.md [hunk 1]
```diff
@@ -22,6 +24,43 @@ torrentHealth(magnet).then(function (res) {
 });
 ```
 
+### Global
```
Comment (dessalines) [inline README.md]: Had to add the global section.

File: cli.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
```
Comment (dessalines) [inline cli.js:1]: The new cli interface.

File: index.js [hunk 3]
```diff
@@ -60,13 +60,18 @@ function read (uri, options, debug) {
 
                 // Remove the "blacklisted" trackers from the list
                 for (var i = 0; i < info.announce.length; i++) {
+                    var announce = info.announce[i];
+                    if (announce.indexOf('/announce') === -1) {
+                        announce += '/announce';
+                    }
+
```
Comment (dessalines) [inline index.js]: This is necessary because the scrape will fail if it doesn't end in announce.

File: index.js [hunk 4]
```diff
@@ -106,16 +114,25 @@ function scrape (req) {
...
+                        name: req.name,
+                        hash: req.hash,
+                        length: req.length,
+                        created: req.created,
                         tracker: trUri,
                         response_time: Date.now() - begin,
                         seeds: data[req.hash].seeders,
-                        peers: data[req.hash].leechers
+                        peers: data[req.hash].leechers,
+                        completed: data[req.hash].completed
```
Comment (dessalines) [inline index.js]: These extra fields are super-useful for a mass scrape process I'm running.

File: index.js [hunk 5]
```diff
@@ -124,30 +141,41 @@ function scrape (req) {
...
+    var created = res[0].created;
 
     for (var i = 0; i < res.length; i++) {
         if (!res[i].error) {
-            totalSeeds += res[i].seeds | 0;
-            totalPeers += res[i].peers | 0;
-            total++;
+            maxSeeds = Math.max(res[i].seeds,maxSeeds);
+            maxPeers = Math.max(res[i].peers,maxPeers);
+            maxCompleted = Math.max(res[i].completed,maxCompleted);
```
Comment (dessalines) [inline index.js]: Makes more sense to use max here, since most scrapes will return zero peers.

File: package.json [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
-    "dependencies": {
-        "torrent-tracker": "^0.0.9",
-        "read-torrent": "^1.3.1"
+  "name": "torrent-tracker-health",
+  "version": "1.0.1",
+  "description": "Get health info about a torrent file or magnet link",
+  "main": "index.js",
+  "bin": {
+    "torrent-tracker-health": "cli.js"
+  },
```
Comment (dessalines) [inline package.json:8]: bin is necessary for the cli.js

**Comment by dessalines** [comment]
I rewrote a lot of this now to support batch processing. I also added unit tests, `bittorrent-tracker`, and the CLI.

**Comment by dessalines** [comment]
It would also fix #3 .

**Comment by vankasteelj** [comment]
Okay sorry it took me so long!

**Comment by vankasteelj** [comment]
I cannot merge this, it makes breaking changes (output isnt backwards compatible), it uses different modules and adds a lot of weight (48Mb instead of 6.5Mb).

I suggest you publish your own module, as there is a need for it (clearly), but I won't accept all those changes here. Seeing how you use bittorrent-tracker instead of torrent-tracker, maybe you can publish your module on NPM using bittorrent-tracker-health name?

**Pull request closed without merging**